itests should not hardcode 0.0.0.0

I recently started seeing this failure on integration tests locally:

```
[WARN] [stderr] io.netty.channel.AbstractChannel$AnnotatedNoRouteToHostException: No route to host: /0.0.0.0:8181
[WARN] [stderr] Caused by: java.net.NoRouteToHostException: No route to host
[WARN] [stderr] 	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
[WARN] [stderr] 	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)
[WARN] [stderr] 	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)
[WARN] [stderr] 	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
[WARN] [stderr] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:707)
[WARN] [stderr] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
[WARN] [stderr] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
[WARN] [stderr] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
[WARN] [stderr] 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
[WARN] [stderr] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[WARN] [stderr] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[WARN] [stderr] 	at java.base/java.lang.Thread.run(Thread.java:829)
```

itests should not assume 0.0.0.0 - this should be configurable, and perhaps should default to `localhost` instead anyway.

--- a/pom.xml
+++ b/pom.xml
@@ -346,8 +346,6 @@
               <argument>--env</argument>
               <argument>CRYOSTAT_WEB_HOST=${cryostat.itest.webHost}</argument>
               <argument>--env</argument>
-              <argument>CRYOSTAT_LISTEN_HOST=${cryostat.itest.webHost}</argument>
-              <argument>--env</argument>
               <argument>CRYOSTAT_RJMX_PORT=9091</argument>
               <argument>--env</argument>
               <argument>CRYOSTAT_WEB_PORT=${cryostat.itest.webPort}</argument>

--- a/pom.xml
+++ b/pom.xml
@@ -34,6 +34,7 @@
   <cryostat.webPort>8181</cryostat.webPort>
   <cryostat.listenPort>9090</cryostat.listenPort>
   <cryostat.itest.webPort>8181</cryostat.itest.webPort>
+  <cryostat.itest.webHost>localhost</cryostat.itest.webHost>
   <cryostat.itest.podName>cryostat-itests</cryostat.itest.podName>
   <cryostat.itest.grafana.port>3000</cryostat.itest.grafana.port>
   <cryostat.itest.grafana.imageStream>quay.io/cryostat/cryostat-grafana-dashboard</cryostat.itest.grafana.imageStream>
@@ -343,9 +344,9 @@
               <argument>--env</argument>
               <argument>CRYOSTAT_DISABLE_SSL=true</argument>
               <argument>--env</argument>
-              <argument>CRYOSTAT_WEB_HOST=0.0.0.0</argument>
+              <argument>CRYOSTAT_WEB_HOST=${cryostat.itest.webHost}</argument>
               <argument>--env</argument>
-              <argument>CRYOSTAT_LISTEN_HOST=0.0.0.0</argument>
+              <argument>CRYOSTAT_LISTEN_HOST=${cryostat.itest.webHost}</argument>
               <argument>--env</argument>
               <argument>CRYOSTAT_RJMX_PORT=9091</argument>
               <argument>--env</argument>
@@ -359,9 +360,9 @@
               <argument>--env</argument>
               <argument>CRYOSTAT_TEMPLATE_PATH=/opt/cryostat.d/templates.d</argument>
               <argument>--env</argument>
-              <argument>GRAFANA_DATASOURCE_URL=http://0.0.0.0:8080</argument>
+              <argument>GRAFANA_DATASOURCE_URL=http://${cryostat.itest.webHost}:8080</argument>
               <argument>--env</argument>
-              <argument>GRAFANA_DASHBOARD_URL=http://0.0.0.0:3000</argument>
+              <argument>GRAFANA_DASHBOARD_URL=http://${cryostat.itest.webHost}:3000</argument>
               <argument>--detach</argument>
               <argument>--rm</argument>
               <argument>${cryostat.imageStream}:${cryostat.imageVersion}</argument>
@@ -380,7 +381,7 @@
               <argument>5m</argument>
               <argument>sh</argument>
               <argument>-c</argument>
-              <argument>until curl -sSk http://0.0.0.0:${cryostat.itest.webPort}/health; do sleep 10; done</argument>
+              <argument>until curl -sSk http://${cryostat.itest.webHost}:${cryostat.itest.webPort}/health; do sleep 10; done</argument>
             </arguments>
           </configuration>
         </execution>

--- a/run.sh
+++ b/run.sh
@@ -31,18 +31,6 @@ if [ -z "$CRYOSTAT_EXT_WEB_PORT" ]; then
     CRYOSTAT_EXT_WEB_PORT="$CRYOSTAT_WEB_PORT"
 fi
 
-if [ -z "$CRYOSTAT_LISTEN_HOST" ]; then
-    CRYOSTAT_LISTEN_HOST="$CRYOSTAT_WEB_HOST"
-fi
-
-if [ -z "$CRYOSTAT_LISTEN_PORT" ]; then
-    CRYOSTAT_LISTEN_PORT=9090;
-fi
-
-if [ -z "$CRYOSTAT_EXT_LISTEN_PORT" ]; then
-    CRYOSTAT_EXT_LISTEN_PORT="$CRYOSTAT_LISTEN_PORT"
-fi
-
 if [ -z "$CRYOSTAT_AUTH_MANAGER" ]; then
     CRYOSTAT_AUTH_MANAGER="io.cryostat.net.NoopAuthManager"
 fi
@@ -69,7 +57,6 @@ if ! podman pod exists cryostat; then
         --hostname cryostat \
         --name cryostat \
         --publish $CRYOSTAT_RJMX_PORT:$CRYOSTAT_RJMX_PORT \
-        --publish $CRYOSTAT_EXT_LISTEN_PORT:$CRYOSTAT_LISTEN_PORT \
         --publish $CRYOSTAT_EXT_WEB_PORT:$CRYOSTAT_WEB_PORT
 fi
 
@@ -91,9 +78,6 @@ podman run \
     -e CRYOSTAT_WEB_HOST=$CRYOSTAT_WEB_HOST \
     -e CRYOSTAT_WEB_PORT=$CRYOSTAT_WEB_PORT \
     -e CRYOSTAT_EXT_WEB_PORT=$CRYOSTAT_EXT_WEB_PORT \
-    -e CRYOSTAT_LISTEN_HOST=$CRYOSTAT_LISTEN_HOST \
-    -e CRYOSTAT_LISTEN_PORT=$CRYOSTAT_LISTEN_PORT \
-    -e CRYOSTAT_EXT_LISTEN_PORT=$CRYOSTAT_EXT_LISTEN_PORT \
     -e CRYOSTAT_AUTH_MANAGER=$CRYOSTAT_AUTH_MANAGER \
     -e CRYOSTAT_CONFIG_PATH="/opt/cryostat.d/conf.d" \
     -e CRYOSTAT_ARCHIVE_PATH="/opt/cryostat.d/recordings.d" \

--- a/src/test/java/itest/ApiListingIT.java
+++ b/src/test/java/itest/ApiListingIT.java
@@ -44,6 +44,7 @@ import io.vertx.core.buffer.Buffer;
 import io.vertx.core.json.JsonObject;
 import io.vertx.ext.web.client.HttpRequest;
 import itest.bases.StandardSelfTest;
+import itest.util.Utils;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
 import org.junit.jupiter.api.BeforeEach;
@@ -83,7 +84,8 @@ public class ApiListingIT extends StandardSelfTest {
                 response.getJsonObject("data").getJsonObject("result"), Matchers.notNullValue());
         MatcherAssert.assertThat(
                 response.getJsonObject("data").getJsonObject("result").getString("overview"),
-                Matchers.equalTo("http://0.0.0.0:8181/HTTP_API.md"));
+                Matchers.equalTo(
+                        String.format("http://%s:%d/HTTP_API.md", Utils.WEB_HOST, Utils.WEB_PORT)));
         MatcherAssert.assertThat(
                 response.getJsonObject("data").getJsonObject("result").getJsonArray("endpoints"),
                 Matchers.notNullValue());

--- a/src/test/java/itest/AutoRulesIT.java
+++ b/src/test/java/itest/AutoRulesIT.java
@@ -52,6 +52,7 @@ import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
 import itest.bases.ExternalTargetsTest;
 import itest.util.Podman;
+import itest.util.Utils;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
 import org.junit.jupiter.api.AfterAll;
@@ -282,11 +283,19 @@ class AutoRulesIT extends ExternalTargetsTest {
         MatcherAssert.assertThat(
                 recording.getString("downloadUrl"),
                 Matchers.equalTo(
-                        "http://0.0.0.0:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Fcryostat-itests:9093%2Fjmxrmi/recordings/auto_Auto_Rule"));
+                        "http://"
+                                + Utils.WEB_HOST
+                                + ":"
+                                + Utils.WEB_PORT
+                                + "/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Fcryostat-itests:9093%2Fjmxrmi/recordings/auto_Auto_Rule"));
         MatcherAssert.assertThat(
                 recording.getString("reportUrl"),
                 Matchers.equalTo(
-                        "http://0.0.0.0:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Fcryostat-itests:9093%2Fjmxrmi/reports/auto_Auto_Rule"));
+                        "http://"
+                                + Utils.WEB_HOST
+                                + ":"
+                                + Utils.WEB_PORT
+                                + "/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Fcryostat-itests:9093%2Fjmxrmi/reports/auto_Auto_Rule"));
     }
 
     @Test

--- a/src/test/java/itest/BasicCommandChannelIT.java
+++ b/src/test/java/itest/BasicCommandChannelIT.java
@@ -87,7 +87,8 @@ public class BasicCommandChannelIT extends StandardSelfTest {
         assertResponseStatus(resp);
         String url = resp.getString("payload");
         MatcherAssert.assertThat(
-                url, Matchers.equalTo(String.format("http://0.0.0.0:%d", Utils.WEB_PORT)));
+                url,
+                Matchers.equalTo(String.format("http://%s:%d", Utils.WEB_HOST, Utils.WEB_PORT)));
     }
 
     @Test

--- a/src/test/java/itest/NotificationsUrlIT.java
+++ b/src/test/java/itest/NotificationsUrlIT.java
@@ -119,7 +119,7 @@ public class NotificationsUrlIT extends StandardSelfTest {
                 future.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS),
                 Matchers.equalTo(
                         String.format(
-                                "{\"notificationsUrl\":\"ws://0.0.0.0:%d/api/v1/notifications\"}",
-                                Utils.WEB_PORT)));
+                                "{\"notificationsUrl\":\"ws://%s:%d/api/v1/notifications\"}",
+                                Utils.WEB_HOST, Utils.WEB_PORT)));
     }
 }

--- a/src/test/java/itest/util/Utils.java
+++ b/src/test/java/itest/util/Utils.java
@@ -46,9 +46,11 @@ import io.vertx.ext.web.client.WebClient;
 public class Utils {
 
     public static final int WEB_PORT;
+    public static final String WEB_HOST;
 
     static {
         WEB_PORT = Integer.valueOf(System.getProperty("cryostatWebPort"));
+        WEB_HOST = System.getProperty("cryostatWebHost", "localhost");
     }
 
     public static final HttpClientOptions HTTP_CLIENT_OPTIONS;
@@ -59,7 +61,7 @@ public class Utils {
                         .setSsl(false)
                         .setTrustAll(true)
                         .setVerifyHost(false)
-                        .setDefaultHost("0.0.0.0")
+                        .setDefaultHost(WEB_HOST)
                         .setDefaultPort(WEB_PORT)
                         .setLogActivity(true);
     }

